### PR TITLE
Add Prometheus & Grafana monitoring to the Ansible deployment (Rebased)

### PR DESCRIPTION
**Pull request by stvoutsin**
Changes to enable Prometheus & Grafana monitoring for the Ansible based Hadoop/Spark deploy.
This PR replaces https://github.com/wfau/aglais/pull/457 and includes more recent changes to the wfau/master branch

### CODE REVIEW

File: deployments/hadoop-yarn/ansible/roles/prometheus/vars/main.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+serviceName: "prometheus"
+userId: "prometheus"
+groupId: "prometheus"
+exec_command: "/usr/local/bin/prometheus --config.file=/etc/prometheus/prometheus.conf --storage.tsdb.path=/data/prometheus --storage.tsdb.retention=2d"
+version: "2.3.2"
```
Comment (Zarquan) [inline deployments/hadoop-yarn/ansible/roles/prometheus/vars/main.yml:6]: Why are we installing an old version ?
https://github.com/prometheus/prometheus/releases
https://github.com/prometheus/prometheus/releases/latest
```suggestion
version: "2.26.0"
```

File: deployments/hadoop-yarn/ansible/roles/prometheus_node_exporter/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+serviceName: "node_exporter"
+userId: "node_exporter"
+groupId: "node_exporter"
+exec_command: /usr/local/bin/node_exporter
+version: 0.16.0
```
Comment (Zarquan) [inline deployments/hadoop-yarn/ansible/roles/prometheus_node_exporter/vars/main.yml:6]: Why are we installing an old version ?
https://github.com/prometheus/node_exporter/releases
https://github.com/prometheus/node_exporter/releases/latest
```suggestion
version: 1.1.2
```

**Review by Zarquan — CHANGES_REQUESTED** [review]
Prefix variables with the role name.
Use the latest version wherever possible. If the latest version doesn't work, then need to add a comment describing why.

**Comment by stvoutsin** [comment]
I've accepted the suggestions, made some minor fixes (typos to var names), and run a test deploy successfully. I did not run any Spark jobs in the test deploy, just checked the Grafana Dashboard to see that the node metrics are getting harvested.

**Comment by Zarquan** [comment]
> I did not run any Spark jobs in the test deploy, just checked the Grafana Dashboard to see that the node metrics are getting harvested.
It would be good to run at least one real world Spark example, the RandomForest is the one I've been using, just to check that (a) this doesn't break Spark for some unknown reason, and (b) we can get some useful metrics about a Spark run out of it.

I have no idea how this PR could break Spark or Zeppelin, but we should check it before we merge, just in case.

Until we have an automated testing in place I think we should consider a set of notes describing a manual run of the RandomForest notebook the minimal acceptance test for a merge.

**Review by Zarquan — APPROVED** [review]

**Pull request merged**